bot/src/modules/birthday/commands/birthday/set.ts

```diff
@@ -32,124 +52,96 @@ export default SubCommand(
+
+    if (hasBirthday && hasBirthday.timezone !== null && !isSameDate && isPastLockout) {
       failEmbed.setDescription(
         "You can only change your birthday once a year, contact an admin if there was a mistake",
       );
       return { embeds: [failEmbed] };
     }
 
-    const query = await state.db.birthday.upsert({
-      where: {
-        userId_guildId: {
-          userId: msg.user.id,
-          guildId: msg.guildId,
-        },
-      },
+    // Upsert the birthday
+    const query = await state.db.user.upsert({
+      where: { id: msg.user.id },
       create: {
-        date: birthdayMoment.toDate(),
-        user: connectOrCreate(msg.user.id),
-        guild: {
-          connect: { id: msg.guild?.id },
+        id: msg.user.id,
+        birthdate: dateNoZone.toJSDate(),
+        timezone: zone,
+        birthdayUpdatedAt: new Date(),
+        UserGuildConfig: {
+          connectOrCreate: {
+            where: { userId_guildId: { guildId: msg.guild.id, userId: msg.user.id } },
+            create: { guildId: msg.guild.id, birthdayEnabled: true },
+          },
         },
       },
       update: {
-        date: birthdayMoment.toDate(),
-        updatedAt: undefined,
+        birthdate: dateNoZone.toJSDate(),
+        birthdayUpdatedAt: new Date(),
+        timezone: zone,
+        UserGuildConfig: {
+          connectOrCreate: {
+            where: { userId_guildId: { guildId: msg.guild.id, userId: msg.user.id } },
+            create: { guildId: msg.guild.id, birthdayEnabled: true },
+          },
+        },
       },
     });
-    if (!query.date) throw new Error("No date???");
-
-    const nextBirthday = nextDate(query.date);
-    const age = yearsAgo(birthdayMoment.toDate());
-    const starSign = getStarSign(query.date);
 
     embed.setTitle("Birthday set!");
+    if (query.birthdate === null || !query.timezone) throw new Error("No date???");
+    embed = formatBirthdayEmbed(embed, { birthdate: query.birthdate, timezone: query.timezone });
+    embed.setThumbnail(getAvatar(msg.member ?? msg.user) || null);
+
     embed.addFields([
       {
-        name: `Birth Date`,
-        value: `**you were born on** ${birthdayMoment.format(
-          "DD-MM-YYYY",
-        )}`,
-      },
-      {
-        name: `Info`,
-        value:
-          `**Age:** ${age} years \n` +
-          `**Next birthday:** <t:${Number(nextBirthday) / 1000}:R>`,
-        inline: true,
-      },
-      {
-        name: `Star Sign`,
-        value: `${starSign?.name} ${starSign?.icon}`,
-        inline: true,
-      },
-      {
-        name: "Disclaimer",
-        value: "All times are recorded in UTC timezone. The “next birthday” and birthday role times may be inaccurate due to this.",
+        name: "Note",
+        value: `You have **${lockoutMinutes} minutes** to correct any mistakes, after that you will have to wait a year to change it again.\n use \`/birthday toggle\` in any other server you'd like to enable birthday notifications in.`,
```

Comment:
The `remove` command doesn't seem to handle this case, I think removing your birthday before the lockout should allow the user to then be allowed to set a birthday again with no time limit.